processor/src/main/java/org/mapstruct/ap/internal/model/common/TypeFactory.java

```diff
@@ -261,7 +261,7 @@ else if ( type.isPrimitive() ) {
      */
     public ExecutableType getMethodType(TypeElement includingType, ExecutableElement method) {
         DeclaredType asType = (DeclaredType) replaceTypeElementIfNecessary( elementUtils, includingType ).asType();
-        TypeMirror asMemberOf = typeUtils.asMemberOf( asType, method );
+        TypeMirror asMemberOf = SpecificCompilerWorkarounds.asMemberOf( asType, method, typeUtils, elementUtils );
```

Comment:
Hm, yes, I guess we could. But I also like Sjaaks idea to wrap the `Types` and `Elements` instances with delegators, then we can't miss anything anymore.